enhancements/local-storage/automatic-detection.md

```diff
@@ -64,40 +65,308 @@ and we can use that information to sort devices into storage classes based on th
+  // LocalVolumeDiscovery Daemon
+  // +optional
+  Tolerations []corev1.Toleration `json:"tolerations,omitempty"`
+}
+
+type LocalVolumeDiscovery struct {
+  metav1.TypeMeta   `json:",inline"`
+  metav1.ObjectMeta `json:"metadata"`
+  Spec              LocalVolumeDiscoverySpec   `json:"spec"`
+  Status            LocalVolumeDiscoveryStatus `json:"status,omitempty"`
+}
+
+type LocalVolumeDiscoveryList struct {
+  metav1.TypeMeta `json:",inline"`
+  metav1.ListMeta `json:"metadata"`
+  Items           []LocalVolumeDiscovery `json:"items"`
+}
+```
+
+API scheme for `LocalVolumeDiscoveryResult` CR:
+```go
+
+// DeviceState defines the observed state of the disk
+type DeviceState string
+
+const (
+  // DeviceAvailable means that the device is available to use and a new persistent volume can be provisioned on it
+  DeviceAvailable DeviceState = "Available"
+  // DeviceProvisioned means that a persistent volume is already provisioned on this device
+  DeviceProvisioned DeviceState = "Provisioned"
+  // DeviceNotAvailable means that the device is already used by some other process and can't be used to provision a Persistent Volume
+  DeviceNotAvailable DeviceState = "NotAvailable"
+)
+
+// DeviceStatus defines the observed state of B
+type DeviceStatus struct{
+  // State is the current state of the block device (Available/In-Use)
+  State DeviceState `json:"state"`
+  // ProvisionedPV is the persisent volume provisioned on the device
+  ProvisionedPV string`json:"provisionedPV,omitempty"`
+  // StorageClass of the provisioned Persistent Volume
+  StorageClass string `json:"storageClass,omitempty"`
+}
+
+// DiscoveredDevice represents the properties of the discovered devices
+type DiscoveredDevice struct {
+  // Name represents the persistent name of the device. For eg, /dev/disk/by-id/...
+  Name string `json:"name"`
+  // Model of the discovered device
+  Model string `json:"model,omitempty"`
+  // Type of the discovered device
+  Type DeviceType `json:"type"`
+  // Vendor of the discovered device
+  Vendor string `json:"vendor,omitempty"`
+  // Size of the discovered device
+  Size resource.Quantity `json:"size,omitempty"`
+  // Property represents whether the device type is rotational or not
+  Property DeviceMechanicalProperty `json:"property"`
+  // FSType represents the filesystem available on the device
+  FSType string `json:"fstype",omitempty`
+  // Status defines whether the device is available for use or not
+  Status DeviceStatus `json:"status",omitempty`
+}
+
+type LocalVolumeDiscoveryResultSpec struct {
+}
+
+type LocalVolumeDiscoveryResultStatus struct {
+  // Conditions is a list of conditions and their status.
+  Conditions          []operatorv1.OperatorCondition `json:"conditions,omitempty"`
+  DiscoveredTimeStamp string                         `json:"discoveredTimeStamp,omitempty"`
+  // NodeName represent the node for which LocalVolumeDiscoveryResult was created
+  NodeName string `json:"nodeName"`
```

Comment:
I think `NodeName` can be moved to spec because at the time this object is created, nodeName is basically already known. `NodeName` is not result of some reconciliation.